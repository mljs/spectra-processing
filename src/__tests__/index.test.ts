import * as SpectraProcessing from '..';

describe('test existence of functions', () => {
  it('x', () => {
    expect(SpectraProcessing.xFindClosestIndex).toBeInstanceOf(Function);
  });

  it('xy', () => {
    expect(SpectraProcessing.xyIntegral).toBeInstanceOf(Function);
    expect(SpectraProcessing.xyIntegration).toBeInstanceOf(Function);
  });
});
