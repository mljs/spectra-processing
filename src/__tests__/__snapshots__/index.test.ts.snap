// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test existence of exported functions 1`] = `
[
  "reimAbsolute",
  "reimAutoPhaseCorrection",
  "reimFFT",
  "reimPhaseCorrection",
  "getOutputArray",
  "xAbsolute",
  "xAbsoluteMedian",
  "xAbsoluteSum",
  "xAdd",
  "xApplyFunctionStr",
  "xAutoCorrelation",
  "xBoxPlot",
  "xCheck",
  "xCheckLengths",
  "xCorrelation",
  "xCostMatrix",
  "xCrossCorrelation",
  "xCumulative",
  "xDivide",
  "xDotProduct",
  "xEnsureFloat64",
  "xFindClosestIndex",
  "xGetFromToIndex",
  "xGetTargetIndex",
  "xHilbertTransform",
  "xHistogram",
  "xIsEquallySpaced",
  "xIsMonotonic",
  "xMassCenterVectorSimilarity",
  "xMaxAbsoluteValue",
  "xMaxIndex",
  "xMaxValue",
  "xMean",
  "xMeanAbsoluteError",
  "xMeanSquaredError",
  "xMeanWeighted",
  "xMedian",
  "xMedianAbsoluteDeviation",
  "xMinIndex",
  "xMinMaxDelta",
  "xMinMaxValues",
  "xMinValue",
  "xMode",
  "xMultiply",
  "xNoiseSanPlot",
  "xNoiseStandardDeviation",
  "xNorm",
  "xNormed",
  "xPadding",
  "xParetoNormalization",
  "xRescale",
  "xRolling",
  "xRollingAverage",
  "xRollingMax",
  "xRollingMedian",
  "xRollingMin",
  "xRotate",
  "xSampling",
  "xSequentialFillFromStep",
  "xSequentialFillFromTo",
  "xSortAscending",
  "xSortDescending",
  "xStandardDeviation",
  "xSubtract",
  "xSum",
  "xUniqueSorted",
  "xVariance",
  "xyAlign",
  "xyCheck",
  "xyCovariance",
  "xyCumulativeDistributionStatistics",
  "xyEnsureFloat64",
  "xyEnsureGrowingX",
  "xyEquallySpaced",
  "xyExtract",
  "xyFilter",
  "xyFilterMinYValue",
  "xyFilterTopYValues",
  "xyFilterX",
  "xyFilterXPositive",
  "xyFindClosestPoint",
  "xyGetNMaxY",
  "xyGrowingX",
  "xyIntegral",
  "xyIntegration",
  "xyJoinX",
  "xyMassCenter",
  "xyMassCenterVector",
  "xyMaxClosestYPoint",
  "xyMaximaY",
  "xyMaxMerge",
  "xyMaxY",
  "xyMaxYPoint",
  "xyMedian",
  "xyMergeByCentroids",
  "xyMinClosestYPoint",
  "xyMinimaY",
  "xyMinYPoint",
  "xyPeakInfo",
  "xyRealMaxYPoint",
  "xyRealMinYPoint",
  "xyReduce",
  "xyRolling",
  "xySetYValue",
  "xySortX",
  "xyToXYArray",
  "xyToXYObject",
  "xyUniqueX",
  "xyWeightedMerge",
  "xy2ToXY",
  "xreimSortX",
  "xreimZeroFilling",
  "xyArrayAlign",
  "xyArrayAlignToFirst",
  "xyArrayMerge",
  "xyArrayWeightedMerge",
  "xyObjectBestPoints",
  "xyObjectCheck",
  "xyObjectJoinX",
  "xyObjectMaxXPoint",
  "xyObjectMaxYPoint",
  "xyObjectMinXPoint",
  "xyObjectMinYPoint",
  "xyObjectSlotX",
  "xyObjectSortX",
  "xyObjectSumY",
  "xyObjectToXY",
  "zonesNormalize",
  "zonesWithPoints",
  "matrixAbsoluteMedian",
  "matrixApplyNumericalEncoding",
  "matrixAutoCorrelation",
  "matrixBoxPlot",
  "matrixCenterZMean",
  "matrixCheck",
  "matrixClone",
  "matrixColumnsCorrelation",
  "matrixCreateEmpty",
  "matrixHistogram",
  "matrixMaxAbsoluteZ",
  "matrixMedian",
  "matrixMinMaxAbsoluteZ",
  "matrixMinMaxZ",
  "matrixNoiseStandardDeviation",
  "matrixNumericalDecoding",
  "matrixNumericalEncoding",
  "matrixPQN",
  "matrixSetSubMatrix",
  "matrixToArray",
  "matrixZPivotRescale",
  "matrixZRescale",
  "matrixZRescalePerColumn",
  "createNumberArray",
  "createDoubleArray",
  "createFromToArray",
  "createRandomArray",
  "createStepArray",
  "getRescaler",
  "isPowerOfTwo",
  "nextPowerOfTwo",
  "stringify",
]
`;
